### DISCUSSION TITLE
Non esp packets

### DISCUSSION THREAD

**Discussion by satishkrishna687** [strongswan/strongswan#1585] [General]
I have host to host connection between two machines running ubuntu. I have noticed "NONESP-encap" packets when I configure dpd_delay, rekeying and reauthentication mechanism. Is this exchange of non esp packets during all these normal? Or does it poes any security concerns.

**Comment by tobiasbrunner** [comment]
When a NAT is detected, [NAT traversal / UDP-encapsulation for ESP](https://docs.strongswan.org/docs/5.9/features/natTraversal.html) is activated. To distinguish IKE packets from UDP-encapsulated ESP packets, which are both sent to port 4500, the former are marked with a 4-byte all-zero non-ESP marker that immediately follows the UDP header, which is where the SPI is located in ESP packets.
